### PR TITLE
Android Studio 3 compatibility

### PR DESCRIPTION
**Pull request by MisterY**
A few things to make the project build with Android Studio 3:

-  updating the build tools
- adding flavor dimensions
- handling output file name
- updating Gradle plugin & Gradle

### CODE REVIEW

**Comment by MisterY** [comment]
This is needed in order to proceed with other additions but I decided to send it separately. It is not directly related to other changes I'd like to make.

File: app/build.gradle [hunk 1]
```diff
@@ -49,9 +49,9 @@ android {
     }
 
     applicationVariants.all { variant ->
-        variant.outputs.each { output ->
-            output.outputFile = new File(
-                    output.outputFile.parent, "GnucashAndroid_v${variant.versionName}.apk")
+        variant.outputs.all { output ->
+            //output.outputFile = new File(output.outputFile.parent, "GnucashAndroid_v${variant.versionName}.apk")
```
Comment (rivaldi8) [inline app/build.gradle]: Unless there's some good reason to leave the old line commented, please, remove it. Git already takes care of remembering it.

File: app/build.gradle [hunk 2]
```diff
@@ -104,6 +106,8 @@ android {
             resValue "string", "dropbox_app_key", "dhjh8ke9wf05948"
             resValue "string", "dropbox_app_secret", "h2t9fphj3nr4wkw"
             resValue "string", "manifest_dropbox_app_key", "db-dhjh8ke9wf05948"
+
+            dimension "stability"
```
Comment (rivaldi8) [inline app/build.gradle]: When there's just one dimension, it's assigned by default to all flavors, you don't need to specify it. See [here](https://google.github.io/android-gradle-dsl/current/com.android.build.gradle.AppExtension.html#com.android.build.gradle.AppExtension:flavorDimensions(java.lang.String[])).

File: app/build.gradle [hunk 3]
```diff
@@ -120,6 +124,8 @@ android {
                 resValue "string", "dropbox_app_secret", "h2t9fphj3nr4wkw"
                 resValue "string", "manifest_dropbox_app_key", "db-dhjh8ke9wf05948"
             }
+
+            dimension "stability"
```
Comment (rivaldi8) [inline app/build.gradle]: Same here. No need to specify it.

File: app/build.gradle [hunk 4]
```diff
@@ -135,6 +141,8 @@ android {
...
```
Comment (rivaldi8) [inline app/build.gradle]: Same here.

**Comment by MisterY** [comment]
Thanks for the feedback! I've applied the changes. Looking forward to making some functional improvements/additions in the near future.

**Pull request merged**